app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx

```diff
@@ -0,0 +1,176 @@
+    if (!selectedDate) return ''
+
+    return selectedDate.toISOString()
+  }, [selectedDate])
+
+  const tzMismatch = useMemo(() => {
+    if (tzOffset === undefined) return
+
+    // Timezone offset in the same format as ActiveSupport
+    const jsTzOffset = new Date().getTimezoneOffset() * -60
+
+    return jsTzOffset !== tzOffset
+  }, [tzOffset])
+
+  const labelIcon = useMemo(() => {
+    if (!tzMismatch) return
+
+    return (
+      <Popover
+        bodyContent={(
+          <p>
+            Your local time zone differs from the provider default.
+            The token will expire at the time you selected in your local time zone.
+          </p>
+        )}
+        headerContent={(
+          <span>Time zone mismatch</span>
+        )}
+      >
+        <button
+          aria-describedby="form-group-label-info"
+          aria-label="Time zone mismatch warning"
+          className="pf-c-form__group-label-help"
+          type="button"
+        >
+          <OutlinedQuestionCircleIcon noVerticalAlign />
+        </button>
+      </Popover>
+    )
+  }, [tzMismatch])
+
+  const handleOnChange = (_value: string, event: FormEvent<HTMLSelectElement>) => {
+    const value = (event.target as HTMLSelectElement).value
+    const selected = collection.find(i => i.id === value) ?? null
+
+    if (selected === null) return
+
+    setDropdownSelectedItem(selected)
+    setCalendarPickedDate(tomorrow)
+  }
+
+  const dateValidator = (date: Date): boolean => {
+    return date >= today
+  }
+
+  return (
+    <>
+      <FormGroup
+        isRequired
+        fieldId={fieldName}
+        helperText={fieldHint}
+        label={fieldLabel}
+        labelIcon={labelIcon}
+      >
+        <FormSelect
+          className="pf-c-form-control-expiration"
+          id={fieldName}
+          value={dropdownSelectedItem.id}
+          onChange={handleOnChange}
+        >
+          {collection.map((item: ExpirationItem) => {
+            return (
+              <FormSelectOption
+                key={item.id}
+                label={item.label}
+                value={item.id}
+              />
+            )
+          })}
+        </FormSelect>
+      </FormGroup>
+      <input id={id} name={id} type="hidden" value={inputDateValue} />
+      {dropdownSelectedItem.id === 'custom' && (
+        <CalendarMonth className="pf-u-mt-md" date={calendarPickedDate} validators={[dateValidator]} onChange={setCalendarPickedDate} />
+      )}
+      {!selectedDate && (
+        <Alert className="pf-u-mt-md" title="Expiration is recommended" variant="warning">
+          It is strongly recommended that you set an expiration date for your token to help keep your information
+          secure
+        </Alert>
+      )}
+    </>
+  )
+}
+
+const ExpirationDatePickerWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ExpirationDatePicker id={props.id} label={props.label} tzOffset={props.tzOffset} />, containerId) }
```

Comment:
You can use prop spread here, it used to be dangerous but Typescript is able to check the type.
```suggestion
const ExpirationDatePickerWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ExpirationDatePicker {...prop} />, containerId) }
```

Reply:
Yes, I saw that in other components, but when I try to do it here I get a linter error: `react/jsx-props-no-spreading`

Reply:
Yeah, that rule was enforced back when we didn't use Typescript but Flow. Type checking was so bad that doing the spread will not check for missing/incorrect props. But Typescript does do that, so feel free to  disable that linter error like it's done in other files.

Reply:
https://github.com/3scale/porta/pull/3943/commits/542518381e121981cafb423b7e2ec10919157a7c